src/arch/wasm/CodeGen.zig

```diff
@@ -3445,60 +3445,19 @@ fn emitUndefined(func: *CodeGen, ty: Type) InnerError!WValue {
     }
 }
 
-/// Returns a `Value` as a signed 32 bit value.
-/// It's illegal to provide a value with a type that cannot be represented
-/// as an integer value.
-fn valueAsI32(func: *const CodeGen, val: Value) i32 {
-    const pt = func.pt;
-    const zcu = pt.zcu;
-    const ip = &zcu.intern_pool;
-
-    switch (val.toIntern()) {
-        .bool_true => return 1,
-        .bool_false => return 0,
-        else => return switch (ip.indexToKey(val.ip_index)) {
-            .enum_tag => |enum_tag| intIndexAsI32(ip, enum_tag.int, pt),
-            .int => |int| intStorageAsI32(int.storage, pt),
-            .ptr => |ptr| {
-                assert(ptr.base_addr == .int);
-                return @intCast(ptr.byte_offset);
-            },
-            .err => |err| @bitCast(ip.getErrorValueIfExists(err.name).?),
-            else => unreachable,
-        },
-    }
-}
-
-fn intIndexAsI32(ip: *const InternPool, int: InternPool.Index, pt: Zcu.PerThread) i32 {
-    return intStorageAsI32(ip.indexToKey(int).int.storage, pt);
-}
-
-fn intStorageAsI32(storage: InternPool.Key.Int.Storage, pt: Zcu.PerThread) i32 {
-    const zcu = pt.zcu;
-    return switch (storage) {
-        .i64 => |x| @as(i32, @intCast(x)),
-        .u64 => |x| @as(i32, @bitCast(@as(u32, @intCast(x)))),
-        .big_int => unreachable,
-        .lazy_align => |ty| @as(i32, @bitCast(@as(u32, @intCast(Type.fromInterned(ty).abiAlignment(zcu).toByteUnits() orelse 0)))),
-        .lazy_size => |ty| @as(i32, @bitCast(@as(u32, @intCast(Type.fromInterned(ty).abiSize(zcu))))),
-    };
-}
-
 fn airBlock(func: *CodeGen, inst: Air.Inst.Index) InnerError!void {
     const ty_pl = func.air.instructions.items(.data)[@intFromEnum(inst)].ty_pl;
     const extra = func.air.extraData(Air.Block, ty_pl.payload);
     try func.lowerBlock(inst, ty_pl.ty.toType(), @ptrCast(func.air.extra[extra.end..][0..extra.data.body_len]));
 }
 
 fn lowerBlock(func: *CodeGen, inst: Air.Inst.Index, block_ty: Type, body: []const Air.Inst.Index) InnerError!void {
-    const pt = func.pt;
-    const wasm_block_ty = genBlockType(block_ty, pt, func.target.*);
-
+    const zcu = func.pt.zcu;
     // if wasm_block_ty is non-empty, we create a register to store the temporary value
-    const block_result: WValue = if (wasm_block_ty != wasm.block_empty) blk: {
-        const ty: Type = if (isByRef(block_ty, pt, func.target.*)) Type.u32 else block_ty;
-        break :blk try func.ensureAllocLocal(ty); // make sure it's a clean local as it may never get overwritten
-    } else .none;
+    const block_result: WValue = if (block_ty.hasRuntimeBitsIgnoreComptime(zcu))
+        try func.allocLocal(block_ty)
```

Comment:
What's the reasoning for this change? When the result never gets written to in a block, it may re-use an existing local with a non-zero value.

Reply:
We dont need clean local when creating placeholder for block. Block instruction that are not `noreturn` have `br`, so value stored in reused local does not matter. Could you provide code snippet that would may result in bad CodeGen?

Reply:
In the past a block result could be returned, but never written to (hence the original comment). It seems this is no longer the case.